compiler/rustc_session/src/session.rs

```diff
@@ -1264,7 +1264,7 @@ fn validate_commandline_args_with_session_available(sess: &Session) {
     // LLVM CFI using rustc LTO requires a single codegen unit.
     if sess.is_sanitizer_cfi_enabled()
         && sess.lto() == config::Lto::Fat
-        && !(sess.codegen_units().as_usize() == 1)
+        && (sess.codegen_units().as_usize() != 1)
```

Comment:
```suggestion
        && sess.codegen_units().as_usize() != 1
```

Why doesn't the `unused_parens` lint trigger here?

Reply:
the lint avoids triggering when mixed-precedence operators are involved because it can be useful to clarify them explicitly. here it's a bit silly though, yeah.